test/test_tensor_creation_ops.py

```diff
@@ -702,6 +702,10 @@ def test_cat_different_devices(self, devices):
                                     "input tensors must be on the same device"):
             torch.cat((cpu, cuda0))
 
+        with self.assertRaisesRegex(RuntimeError,
+                                    "All input tensors and out must be on the same device"):
+            torch.cat((cpu, cpu), out=cuda0)
```

Comment:
In this test it should be cat(cuda0, cuda0) with out=cuda1, and then a SEPARATE test that is @onlyCUDA should test cat(cpu, cpu) with out=cuda

Reply:
Also update that separate test with a call to stack directly to be 100% we're solving the issue. And - who knows - maybe in the future torch.stack() won't call torch.cat().